nifi-api/src/main/java/org/apache/nifi/registry/flow/FlowRegistryBranch.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *
+ *  * Licensed to the Apache Software Foundation (ASF) under one or more
+ *  * contributor license agreements.  See the NOTICE file distributed with
+ *  * this work for additional information regarding copyright ownership.
+ *  * The ASF licenses this file to You under the Apache License, Version 2.0
+ *  * (the "License"); you may not use this file except in compliance with
+ *  * the License.  You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package org.apache.nifi.registry.flow;
+
+import java.util.Objects;
+
+public class FlowRegistryBranch {
```

Comment:
Currently there wasn't a need to know this since I wasn't envisioning any advanced features like merging back or rebasing etc from NiFi. All of these things would be done outside in GitHub, or whatever registry.

Reply:
Yeah, that makes sense. My reasons to ask this was a bit different:
- It can be potentially an useful information for the users to show (it might help follow the lineage in the UI, without hopping to Git or other backend system)
- With this question I wanted to validate my assumption with the code within NiFi does not have to do any kind of work with figuring out what the given branch contains (like from berofe and after branching off, wher the before is shared with the parent) but always expected to get a snapshot view of the actual content)

Reply:
I think this ok to mark as resolved, but let me know if not